spec/controllers/api/v1/providers_controller_spec.rb

```diff
@@ -1,66 +1,146 @@
 require 'rails_helper'
 
 describe API::V1::ProvidersController, type: :controller do
-  describe "index" do
-    it "calls limit on the model with default value of 100" do
-      allow(controller).to receive(:authenticate)
-      expect(Provider).to receive_message_chain(:changed_since, :limit).with(100).and_return([])
+  describe '#index' do
+    RSpec::Matchers.define :have_providers do |*providers|
```

Comment:
I just realised that this is a duplication of what's in `spec/requests/api/v1/provider_spec.rb`, so it should be moved into `spec/support` and included into `spec-helper.rb`.

Reply:
separate PR to get this out?